support-models/src/main/scala/com/gu/support/workers/states/SendThankYouEmailState.scala

```diff
@@ -13,13 +13,12 @@ case class SendThankYouEmailState(
   user: User,
   giftRecipient: Option[GiftRecipient],
   product: ProductType,
-  paymentMethod: Either[PaymentMethod, RedemptionData],
+  paymentOrRedemptionData: Either[PaymentMethodWithSchedule, RedemptionData],
```

Comment:
Definitely agree with renaming this field, however I think we'll need to support the old schema in the deserialiser because we have 2 separate deploys for support-frontend and support-workers

Reply:
this is only between two states of support workers though so I think it would be ok.
I did think about that if there are checkouts in flight at that moment, but I decided to leave it.  Worse case we will get a failed email if we are really unlucky, we can trigger the lambda manually.
Does that make sense?

Reply:
Oh yes you're right 👍

Reply:
so in the end you were right, but not in the way you thought!
Two things
1. SendThankyouEmailState and SendAcquisitionEventState both rely on the same fields, so everything seemed to work but the lambda was actually looping away in the background.  I managed to fix that beforing finding out that....
1. ... FailureHandler relies on paymentFields OR paymentMethod (although all it really needs is the PaymentProvider).  I think the solution would be to add that to the state and pass it through, rather than adding a third special case.

The fix should be independent of this PR so I will do a third PR (tomorrow), to go before this one, with the fixes, to keep it clear and easy to review!

Reply:
for the record the third PR was here: https://github.com/guardian/support-frontend/pull/2577